application/app/schedule.tsx

```
@@ -0,0 +1,127 @@
+import React, { useEffect, useState } from "react";
+import { View, Text, StyleSheet, ScrollView, TouchableOpacity, Alert } from "react-native";
+import { GoogleSignin } from "@react-native-google-signin/google-signin";
+
+export default function Schedule() {
+  const [events, setEvents] = useState([]);
+  const [loading, setLoading] = useState(true);
+
+  const timeSlots = Array.from({ length: 24 }, (_, i) => {
+    const hour = i;
+    const minutes = i % 2 === 0 ? "00" : "30";
+    return `${hour}:${minutes}`; 
+  });
+
+  const getCurrentWeek = () => {
+    const today = new Date();
+    const startOfWeek = new Date(today);
+    startOfWeek.setDate(today.getDate() - today.getDay() + 1);
+    const weekDays = [];
+
+    for (let i = 0; i < 7; i++) {
+      const date = new Date(startOfWeek);
+      date.setDate(startOfWeek.getDate() + i);
+      weekDays.push({
+        day: date.toLocaleDateString("en-US", { weekday: "short" }),
+        date: date.getDate(),
+        fullDate: date.toISOString(),
+      });
+    }
+    return weekDays;
+  };
+
+  const weekDays = getCurrentWeek();
+
+  const fetchCalendarEvents = async (accessToken) => {
```

Im wondering what would happen if there is a cases when there is no events returned to API. Can you inclide another error handling case plz? You can add this code to handle the case 
```
if (!response.ok) { 
    console.error("Error fetching calendar events"); 
    return; 
} 
```

---

This code correctly handles failed responses but could be improved by providing more details in the error log, such as response.status and response.statusText, for better debugging. Consider throwing an error instead of just returning to ensure proper error handling upstream.